core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -252,9 +260,8 @@ public void resetUpdateType()
 
     public boolean isUpdateTarget(Table table)
     {
-        return ("DELETE".equals(updateType) || "UPDATE".equals(updateType)) &&
-                target.orElseThrow(() -> new IllegalStateException("Update target not set"))
-                        .getTable().orElseThrow(() -> new IllegalStateException("Table reference not set in update target")) == table; // intentional comparison by reference
+        return tableToBeUpdated.map(NodeRef.of(table)::equals) // intentional comparison by reference
```

Comment:
you don't need `NodeRef` here.

```
requrieNonNull(table, "table is null");
return tableToBeUpdated.orElseThrow(() -> new IllegalStateException("Update target not set") && 
   tableToBeUpdated.get() == table; // intentional comparison by reference
```

Reply:
We might not throw an exception here, for non UPDATE/DELETE queries this might not be set and so we might return false in that case.